src/Build/BuildCop/Infrastructure/ConfigurationProvider.cs

```diff
@@ -0,0 +1,206 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.Json.Serialization;
+using System.Text.Json;
+using Microsoft.Build.Experimental.BuildCop;
+using System.Configuration;
+
+namespace Microsoft.Build.BuildCop.Infrastructure;
+
+
+// TODO: https://github.com/dotnet/msbuild/issues/9628
+//  Let's flip form statics to instance, with exposed interface (so that we can easily swap implementations)
+internal static class ConfigurationProvider
+{
+    // TODO: This module should have a mechanism for removing unneeded configurations
```

Comment:
Could you please clarify what kind of configuration we want to remove and from where?

Reply:
Sorry for the cryptic note :-)

The idea was that there might be cases where we know that the analyzer cannot be used in the build anymore (e.g. when issue during setup is encounteredand the analyzer is removed in `SetupAnalyzersForNewProject`) - in such case we might not need to cache the configuration for the analyzer anymore.

Though this is not a must have requirement.

Reply:
Got it, thank you! 
In such case we might need to add public api in ConfigurationProvider which will get the information what to remove from local cache. 
Agree on prioritization

Reply:
Btw. - the API should still be internal. The whole `ConfigurationProvider` should not be of concern for users - those should only see an authoring API and OM